pymongo/asynchronous/encryption.py

```diff
@@ -213,6 +213,9 @@ async def kms_request(self, kms_context: MongoCryptKmsContext) -> None:
                     if not data:
                         raise OSError("KMS connection closed")
                     kms_context.feed(data)
+            # Async raises an OSError instead of returning empty bytes
+            except OSError as err:
+                raise OSError("KMS connection closed") from err
```

Comment:
Doesn't OSError already get handled by the `except Exception as exc` block? It should hit the _raise_connection_failure line.

Reply:
Fixed